Sub/Src/Mission/mission_commander.py

```diff
@@ -0,0 +1,310 @@
+        self.mission_file = self.param_serv.get_param("Missions/mission_file")
+        self.mission_live = False
+
+        print("[INFO]: New Mission file set as %s", self.mission_file)
+        #Parse the mission file
+        self.parse_mission()
+    def _command_listener(self):
+        '''
+        The thread to run update requests from the GUI to tell the mission commander
+        when it is ready to run missions and what missions to do.
+
+        Parameters:
+            N/A
+        Returns:
+            N/A
+        '''
+        while self.command_listener_thread_run:
+            try:
+                #Recieve commands from the the GUI and/or Mission Commander
+                self.mission_commander_node.spinOnce(self.movement_mode_subscriber)
+
+            except Exception as e:
+                print("[ERROR]: Could not properly recieved messages in command listener. Error:", e)
+            time.sleep(0.2)
+
+    def parse_mission(self):
+        '''
+        Parse the mission .json file and generate the code for each task. Save
+        the individual task in the mission_tasks list attribute.
+        Parameters:
+            N/A
+        Returns:
+            N/A
+        '''
+
+        with open(self.mission_file, 'r') as f:
+            self.mission_data = json.load(f)
+
+        #Count the number of tasks in the mission
+        self.num_tasks = len(self.mission_data)
+        print("[INFO]: Number of tasks for mission is", self.num_tasks, ".")
+
+        task_keys = self.mission_data.keys()
+
+        for task_index, task in enumerate(task_keys):
+
+            #Get the task type and name
+            task_type = self.mission_data[task]["type"]
+
+            #generate waypoint task
+            if(task_type == "Waypoint"):
+                waypoint_task = Waypoint_Task(self.mission_data[task], self.drive_functions)
+                self.mission_tasks.append(waypoint_task)
+
+    def run(self):
+        '''
+        Run the mission tasks sequentially.
+
+        Parameters:
+            N/A
+        Returns:
+            N/A
+        '''
+        while(self.run_thread):
+
+            try:
+                #If in Mission mode, listen to see if the autonomous mode button is
+                #pressed.
+                """
+                if(self.mission_mode):
+
+                    if(self.auto_serial.in_waiting):
+                        auto_pressed = (self.auto_serial.read(13)).decode()
+                        self.auto_serial.read(2) #Read the excess two bytes
+
+                        if(auto_pressed == "Auto Status:1"):
+                            print("[INFO]: Mission Now Live")
+                            self.mission_live = True
+
+                        elif(auto_pressed == "Auto Status:0"):
+                            print("[INFO]: Mission is no longer Live.")
+                            self.mission_live = False
+                """
+                if(self.mission_mode):
```

Comment:
Remove lines 278 to 282. This was strictly for testing purposes. Move to using the auto switch.